forge/src/gas_report.rs

```diff
@@ -54,7 +54,10 @@ impl GasReport {
         }
 
         if let Some(name) = &trace.contract {
-            let report_for = self.report_for.iter().any(|s| s == name);
+            let report_for = self
+                .report_for
+                .iter()
+                .any(|s| s == name.clone().rsplit(':').next().unwrap_or(name));
```

Comment:
```suggestion
            let report_for = self
                .report_for
                .iter()
                .any(|s| s == name.rsplit(':').next().unwrap_or(name.as_str()));
```
let's also add a brief note here